### ISSUE TITLE
Do not check whether an ingredient specification corresponds to the "amount unit ingredient" syntax

### ISSUE THREAD

**Issue by cloud2018** [nextcloud/cookbook#1957]
**Description**
Cookbook checks whether the syntax "amount unit ingredient" is used for ingredients. 
In many (real) recipes, some ingredients are specified without an exact quantity. This is often the case with salt, pepper or optional ingredients. Even schem.org does not require such a syntax.

In the ingredient list, the app's message is very confusing if ingredients are not specified with amount and unit.

In my opinion, it would be better not to check and report "... Please change it to this syntax: amount unit ingredient ..." or to be able to switch this off in the settings.

**Screenshots**

![nc_cb_02](https://github.com/nextcloud/cookbook/assets/34692335/734b40da-0587-428c-b5a6-8610b9ff1af5)

**Versions**
Nextcloud server version: 26.0.8
Cookbook version: 0.10.4

**Comment by christianlupus** [comment]
OK,

> [...] it would be better not to check and report "... Please change it to this syntax: amount unit ingredient ..." [...]

What do mean by this? Just replacing the text in case of failed parsing? I am open for any good suggestion.

Adding an option to hide this in the settings might be doable. Any other considerations to be made?

**Comment by cloud2018** [comment]
I would completely dispense with the check and therefore dispense with the "!" next to the ingredient and dispense with the text "The ingredient cannot be recalculated due to incorrect syntax. Please change it to this syntax: amount unit ingredient. Examples: 200 g carrots or 1 pinch of salt" below the list of ingredients

I don't understand that following formulations 

- "salt and pepper"
- "a little sugar to taste"
- "optional sugar"
- "Jam as desired"

are described as incorrect syntax. It is obvious that such specifications are independent of the amount of portions and do not need to be recalculated.

**Comment by christianlupus** [comment]
I am not on the same site here.

The problem is, that we can have multiple cases

1. The ingredient is parsed correctly. No asterisk is added and no warning is shown.
2. The ingredient is non-quantitative (_jam as desired_). Here ideally, the asterisk should be hidden as well. It appears but is just a minor optical burden.
3. The ingredient is quantitative but could not be parsed numerically (_A bunch of spring onions_). Ideally, this could be parsed into _1 bunch of spring onion_ but falls back to the warning you mentioned.
4. The most critical case was if we thought we interpreted correctly but failed and miscaclulated.

The problem with case 3 is that it is really hard to parse this. You have to consider that we might have non-English users, especially think of Arabic (right-to-left) and Asian languages. I am not fluent in many languages and I found no help to solve this problem uniqly for all possible languages on earth.

I also asked the design team of NC if they had a good idea to cope with this issue. The discussion is [reachable in the public chat](https://cloud.nextcloud.com/call/gqff69i8#message_2393926). There was no really good idea coming up. we mainly agreed that just hiding the message might cause more trouble with other users.

**Comment by cloud2018** [comment]
@christianlupus 
Aus Anwendersicht wäre das verständlich, wenn die Warnmeldung nur erscheinen würde, wenn das Rezept initil von einer URL geladen und geparst wurde oder wenn der Portionsrechner eine andere  Portion berechnet und die Meldung nur einmal erscheint. Viele (geschätzt >50%) Rezepte enthalten Formulierungen wie "... zum Abschmecken" oder "Salz und Pfeffer". Da ist die kursive Schriftart, das ! und der zusätzliche Text für die Übersichtlichkeit sehr störend und ungewohnt, da viele Rezepte-Seiten im Web solche Warnhinweise nicht verwenden.
Evtl. wäre es eine Variante, wenn man konfigurieren könnte, ob man die kursive Schriftart, das ! und den zusätzlichen Text angezeigt haben möchte.

Für mich selbst ist dieser Punkt auch nicht mehrt so wichtig, da ich über CSS die neue Darstellung unterdrücke

/* bei Cookbook Fehlermeldung wg. Syntax Menge Einheit Zutat: */ 
.ingredients > ul > li > span.icon-error  {
display:none;}
.ingredients > ul > li.unindented {
font-style: normal !important;
}
.ingredient-parsing-error {
display:none;
}
Zum Schluß ein riesiges Dankeschön für die sehr gute App, sie gehört zu meinen wichtigsten und am häufigsten benutze App bei Nextcloud
